### PR TITLE
Added equality and deep-equality checks for files

### PR DESCRIPTION
**Pull request by BigstickCarpet**
Implemented file equality (compares contents) and deep-equality (compares contents, times, owner, and group) as mentioned on the [planned assertions page](https://github.com/chaijs/chai-fs/blob/master/docs/planned.md)

### CODE REVIEW

**Comment by keithamus** [comment]
LGTM 👍 good work @BigstickCarpet and welcome aboard 😄

**Comment by BigstickCarpet** [comment]
Just realized that I forgot to update `README.md`, and I forgot to remove these assertions from `planned.md`.   I'll make those changes later today, and then we can merge the PR

**Comment by keithamus** [comment]
Sounds good 😄

**Comment by BigstickCarpet** [comment]
Updated `README.md` and `planned.md`

**Comment by BigstickCarpet** [comment]
LGTM 👍

**Comment by meeber** [comment]
Awesome will review after work :D

**Comment by BigstickCarpet** [comment]
@keithamus - I guess it requires another "LGTM" from you, since I made another commit?

File: README.md [hunk 1]
```diff
@@ -125,97 +125,138 @@ Uses `fs.statSync().isFile()`
...
+
+* Reads both files as utf8 text (could update to support base64, binary Buffer etc).
 * To negate this using `expect/should` you chain the `.not`-negation ***after*** the regular `file()`.
-* The `with` chain is just syntax sugar.
 
-### file().using.json.schema(obj)
+### file().and.deep.equal(otherPath)
 
-Assert the path exists, is a file, contains json parsable text conforming to given JSON-Schema.
+Assert the _both_ paths exist, are files, contain the same content, and have the same attributes, including:
```
Comment (meeber) [inline README.md]: Nitpick: "Assert the" should be "Assert that"

File: README.md [hunk 2]
```diff
@@ -125,97 +125,138 @@ Uses `fs.statSync().isFile()`
...
-* Reads file as utf8 text (could update to support base64, binary Buffer etc). 
+* Reads file as utf8 text (could update to support base64, binary Buffer etc).
 
-Note: *In a future version this might be supported as a chain behind file() and directory()* 
+Note: *In a future version this might be supported as a chain behind file() and directory()*
 
 ### content.that.match(/xyz/)
 
-Assert the path exists, is a file and has content that match the regular expression. 
+Assert the path exists, is a file and has content that match the regular expression.
```
Comment (meeber) [inline README.md]: Nitpick: "has content that match" should be "has contents that match"

**Comment by meeber** [comment]
@BigstickCarpet Very nicely done sir. I really like the approach you took implementing the new assertion, and the code is super clean and organized.

It'd be good to squash this PR down into a single commit before merging.

Also, in the future, please separate unrelated linting fixes into a different PR; it made it a bit more difficult to review the relevant changes :D

**Comment by BigstickCarpet** [comment]
@meeber Thanks for the kind words. 

I can squash the commits if that's the preferred practice here.  This seems to vary from project to project on GitHub.  Some teams prefer PRs to contain everything in a single commit; others prefer separate commits for logically-related changes, which is what I did here (one commit for the code changes, one for the tests, one for documentation)

Sorry about all the unrelated linting fixes in the README.md file.  That was unintentional. My editor apparently automatically removed trailing whitespace throughout the file without me realizing it.  I can undo that if you prefer.  LMK

**Comment by meeber** [comment]
No need to undo the linting changes.

Historically, it has been a bit inconsistent across Chai master and the various plugins, but the standard we're striving for is to wrap everything (code, docs, tests) related to a single feature or bug fix in a single commit. But if there were multiple features or bug fixes, each one should be separated into its own commit or PR. This strategy makes it easier for someone reviewing an old change to isolate the commit that introduced the change and everything related to that change, and revert it if needed.

**Comment by BigstickCarpet** [comment]
@meeber I just discovered that as of April of this year, GitHub can [automatically squash commits when merging a PR](https://github.com/blog/2141-squash-your-commits).  The person doing the merge can choose whether to squash or not, depending on the settings of the repo. Pretty cool stuff!

**Comment by meeber** [comment]
@BigstickCarpet That's awesome, thanks for pointing that out. It looks like that squashing feature might need to be enabled in the repo settings, which I don't have access to. @keithamus? 

Regarding the PR: LGTM

**Pull request merged**